src/core/thread/mesh_forwarder.cpp

```diff
@@ -573,12 +574,32 @@ void MeshForwarder::SetRxOnWhenIdle(bool aRxOnWhenIdle)
 
 void MeshForwarder::SetPollPeriod(uint32_t aPeriod)
 {
-    if (mMac.GetRxOnWhenIdle() == false && mPollPeriod != aPeriod)
+    if (!mNetif.IsUp())
```

Comment:
I think this is OK for now, but I'd like to clean this up later.  This component should just keep a single poll period that is used by default.  Other components (e.g. MLE) can maintain their own timers and force a poll when they want, rather than having to set the poll period.  This allows multiple components to request polls without competing on which one is authoritative on the poll period.